app/src/main/java/com/quran/labs/androidquran/ui/util/TranslationsSpinnerAdapter.java

```diff
@@ -87,24 +91,35 @@ public View getView(int position, View convertView, @NonNull ViewGroup parent) {
   @Override
   public View getDropDownView(int position, View convertView, ViewGroup parent) {
     CheckBoxHolder holder;
-    if (convertView == null) {
-      convertView = layoutInflater.inflate(
+    convertView = layoutInflater.inflate(
           R.layout.translation_ab_spinner_item, parent, false);
-      convertView.setTag(new CheckBoxHolder(convertView));
-    }
-    holder = (CheckBoxHolder) convertView.getTag();
+    convertView.setTag(new CheckBoxHolder(convertView));
 
+    holder = (CheckBoxHolder) convertView.getTag();
     holder.checkBoxPosition = position;
     holder.checkBox.setText(translationNames[position]);
     holder.checkBox.setChecked(selectedItems.contains(translations.get(position).filename));
     holder.checkBox.setOnClickListener(onCheckedChangeListener);
+
     return convertView;
   }
 
+  @Override
+  public int getItemViewType(int position) {
+    if (position == translationNames.length - 1){
+      return 1; // Last item in spinner should be text "More Translations"
+    }else{
+      return 0;
+    }
+  }
+
   public void updateItems(String[] translationNames,
                           List<LocalTranslation> translations,
                           Set<String> selectedItems) {
     clear();
+    List<String> transList = new ArrayList<String>(Arrays.asList(translationNames));
```

Comment:
if you'll do that multiple times, extract out the code i showed above in a method